src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+    let iteration = 0;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+      iteration = 1;
+    }
+
+    for (iteration; iteration < this.length; iteration++) {
+      accumulator = callback(accumulator, this[iteration], iteration, this);
+    }
```

Comment:
We on `Array iteration methods` topic. So maybe should use some array method instead of a simple loop )